tszip/compression.py

```diff
@@ -95,8 +97,13 @@ def compress(ts, path, variants_only=False):
         with zarr.ZipStore(filename, mode="w") as store:
             root = zarr.group(store=store)
             compress_zarr(ts, root, variants_only=variants_only)
-        os.replace(filename, destination)
-    logging.info(f"Wrote {destination}")
+        if stdout:
```

Comment:
Ah, looks like we're actually writing a temporary file anyway, so it's not that bad. How about this (untested!)

```python
if isinstance(destination, (str, pathlib.Path)):
     os.replace(filename, destination)
     logging.info(f"Wrote {destination}")
else:
     # Assume that destination is a file-like object open in "wb" mode.
     with open(filename, "rb") as source:
            chunk_size = 2**10  # 1MiB
            for chunk in iter(functools.partial(source.read, 64), b""):
                  destination.write(chunk)
```

That way we don't need the stdout option and can write to any suitable file-like object.

Reply:
In general I like this solution as it helps keep the function signature the same. I think it takes a little bit of messing with because it does not reliably work when constructing an in-place directory when the path being fed in is `sys.stdout`, but I think that it could be worked around. I'll see what I can do for this to make it work with the various tests in place as well. 
```
    destination = str(path)
    # Write the file into a temporary directory on the same file system so that
    # we can write the output atomically.
    destdir = os.path.dirname(os.path.abspath(destination))
    with tempfile.TemporaryDirectory(dir=destdir, prefix=".tszip_work_") as tmpdir:
```